<html>
<head>
<style type="text/css" media="all">
@import url(http://www.ensembl.org/css/ensembl.css);
@import url(http://www.ensembl.org/css/content.css);
#page ul li { list-style-type:none; list-style-image: none; margin-left: -2em }
</style>
<title>Healthcheck results for homo sapiens databases</title>
</head>
<body>
<div id='page'><div id='i1'><div id='i2'><div class='sptop'>&nbsp;</div>
<div id='release'>Healthcheck results for homo sapiens databases</div>
<hr>
<div id='related'><div id='related-box'>
<h2>Results by database</h2>
<ul>
<li><a href="#monodelphis_domestica_core_39_3">m_domestica_core_39_3</a></li>
</ul>
<h2>Results by test</h2>
<ul>
<li><a href="#Accession">Accession</a></li>
<li><a href="#AnalysisDescription">AnalysisDescription</a></li>
<li><a href="#AssemblyException">AssemblyException</a></li>
<li><a href="#AssemblyMultipleOverlap">AssemblyMultipleOverlap</a></li>
<li><a href="#AssemblySeqregion">AssemblySeqregion</a></li>
<li><a href="#AssemblyTablesAcrossSpecies">AssemblyTablesAcrossSpecies</a></li>
<li><a href="#AttribTypeAcrossSpecies">AttribTypeAcrossSpecies</a></li>
<li><a href="#BigGeneExon">BigGeneExon</a></li>
<li><a href="#Biotypes">Biotypes</a></li>
<li><a href="#BlankInfoType">BlankInfoType</a></li>
<li><a href="#ComparePreviousVersionBiotypes">ComparePreviousVersionBiotyp</a></li>
<li><a href="#ComparePreviousVersionTableRows">ComparePreviousVersionTableR</a></li>
<li><a href="#ComparePreviousVersionXrefs">ComparePreviousVersionXrefs</a></li>
<li><a href="#CompareSchema">CompareSchema</a></li>
<li><a href="#CoordSystemAcrossSpecies">CoordSystemAcrossSpecies</a></li>
<li><a href="#CoreForeignKeys">CoreForeignKeys</a></li>
<li><a href="#DNATablesAcrossSpecies">DNATablesAcrossSpecies</a></li>
<li><a href="#DensityFeatures">DensityFeatures</a></li>
<li><a href="#DisplayLabels">DisplayLabels</a></li>
<li><a href="#DisplayMarkerSynonyms">DisplayMarkerSynonyms</a></li>
<li><a href="#DisplayXref">DisplayXref</a></li>
<li><a href="#DuplicateAssembly">DuplicateAssembly</a></li>
<li><a href="#EmptyTables">EmptyTables</a></li>
<li><a href="#ExonStrandOrder">ExonStrandOrder</a></li>
<li><a href="#ExonTranscriptStartEnd">ExonTranscriptStartEnd</a></li>
<li><a href="#ExternalDBDisplayName">ExternalDBDisplayName</a></li>
<li><a href="#FeatureAnalysis">FeatureAnalysis</a></li>
<li><a href="#FeatureCoords">FeatureCoords</a></li>
<li><a href="#GOXrefs">GOXrefs</a></li>
<li><a href="#GeneStatus">GeneStatus</a></li>
<li><a href="#GeneType">GeneType</a></li>
<li><a href="#InterproDescriptions">InterproDescriptions</a></li>
<li><a href="#Karyotype">Karyotype</a></li>
<li><a href="#LogicNamesDisplayable">LogicNamesDisplayable</a></li>
<li><a href="#MappingSession">MappingSession</a></li>
<li><a href="#Meta">Meta</a></li>
<li><a href="#MetaCoord">MetaCoord</a></li>
<li><a href="#MetaCrossSpecies">MetaCrossSpecies</a></li>
<li><a href="#OligoMismatches">OligoMismatches</a></li>
<li><a href="#OligoProbes2Genome">OligoProbes2Genome</a></li>
<li><a href="#PredictedXrefs">PredictedXrefs</a></li>
<li><a href="#ProteinFeatureTranslation">ProteinFeatureTranslation</a></li>
<li><a href="#ProteinFeatures">ProteinFeatures</a></li>
<li><a href="#ProteinIDXrefsNoVersion">ProteinIDXrefsNoVersion</a></li>
<li><a href="#Pseudogene">Pseudogene</a></li>
<li><a href="#RepeatConsensus">RepeatConsensus</a></li>
<li><a href="#RepeatFeature">RepeatFeature</a></li>
<li><a href="#SeqRegionAcrossSpecies">SeqRegionAcrossSpecies</a></li>
<li><a href="#SeqRegionAttribAcrossSpecies">SeqRegionAttribAcrossSpecies</a></li>
<li><a href="#SeqRegionCoordSystem">SeqRegionCoordSystem</a></li>
<li><a href="#SeqRegionsTopLevel">SeqRegionsTopLevel</a></li>
<li><a href="#SingleDBCollations">SingleDBCollations</a></li>
<li><a href="#StableID">StableID</a></li>
<li><a href="#Strand">Strand</a></li>
<li><a href="#TranscriptsTranslate">TranscriptsTranslate</a></li>
<li><a href="#TranslationStartEnd">TranslationStartEnd</a></li>
<li><a href="#TranslationStartEndExon">TranslationStartEndExon</a></li>
<li><a href="#XrefTypes">XrefTypes</a></li>
</ul>
</div></div>
<h2>Summary</h2>
<p><strong><font color='green' size=-1>40</font> tests passed and <font color='red' size=-1>11</font> failed out of a total of 51 tests run.</strong></p>
<hr>
<h2>Summary by database</h2>
<p><table class='ss'>
<tr><th>Database</th><th>Passed</th><th>Failed</th></tr>
<tr><td><a href="#monodelphis_domestica_core_39_3"><font color='red' size=-1>monodelphis_domestica_core_39_3</font></a></td><td><font color='green' size=-1>40</font></td><td><font color='red' size=-1>11</font></td></tr>
</table></p>
<hr>
<h2>Summary by test</h2>
<p><table class='ss'>
<tr><th>Test</th><th>Passed</th><th>Failed</th></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Accession"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Accession</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AnalysisDescription"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AnalysisDescription</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyException"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblyException</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyMultipleOverlap"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblyMultipleOverlap</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblySeqregion"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblySeqregion</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.BigGeneExon"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.BigGeneExon</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Biotypes"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Biotypes</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.BlankInfoType"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.BlankInfoType</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CompareSchema"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.CompareSchema</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CoordSystemAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.CoordSystemAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CoreForeignKeys"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.CoreForeignKeys</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DensityFeatures"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DensityFeatures</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayLabels"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DisplayLabels</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayXref"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.DisplayXref</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateAssembly"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DuplicateAssembly</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.EmptyTables"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.EmptyTables</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExonStrandOrder"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ExonStrandOrder</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.FeatureAnalysis"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.FeatureAnalysis</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.FeatureCoords"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.FeatureCoords</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GOXrefs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.GOXrefs</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GeneStatus"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.GeneStatus</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GeneType"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.GeneType</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.InterproDescriptions"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.InterproDescriptions</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Karyotype"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.Karyotype</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MappingSession"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.MappingSession</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Meta"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Meta</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MetaCoord"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.MetaCoord</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.OligoMismatches"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.OligoMismatches</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.OligoProbes2Genome"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.OligoProbes2Genome</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.PredictedXrefs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.PredictedXrefs</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatures"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.ProteinFeatures</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Pseudogene"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Pseudogene</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.RepeatConsensus"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.RepeatConsensus</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.RepeatFeature"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.RepeatFeature</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SingleDBCollations"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SingleDBCollations</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.StableID"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.StableID</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Strand"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Strand</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEnd"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.TranslationStartEnd</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.XrefTypes"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.XrefTypes</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
</table></p>
<hr>
<h2>Detailed reports by database</h2>
<h3 class='boxed'><a name="monodelphis_domestica_core_39_3">monodelphis_domestica_core_39_3</a></h3>
<p>
<a name="monodelphis_domestica_core_39_3:BigGeneExon"></a> <font color='grey'><strong>BigGeneExon: </strong>18 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:ComparePreviousVersionBiotypes"></a> <font color='red'><strong>ComparePreviousVersionBiotypes: </strong>monodelphis_domestica_core_38_2b has 2 biotypes of type Mt-rRNA but monodelphis_domestica_core_39_3 has none</font><br>
<a name="monodelphis_domestica_core_39_3:ComparePreviousVersionBiotypes"></a> <font color='red'><strong>ComparePreviousVersionBiotypes: </strong>monodelphis_domestica_core_38_2b has 21 biotypes of type Mt-tRNA but monodelphis_domestica_core_39_3 has none</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:DisplayXref"></a> <font color='red'><strong>DisplayXref: </strong>7513 gene display_xrefs are set to 0 - should be set to NULL</font><br>
<a name="monodelphis_domestica_core_39_3:DisplayXref"></a> <font color='red'><strong>DisplayXref: </strong>7513 genes (out of a total of 20907) have display_xref_ids pointing to non-existent xrefs.</font><br>
<a name="monodelphis_domestica_core_39_3:DisplayXref"></a> <font color='red'><strong>DisplayXref: </strong>33804 transcript display_xrefs are set to 0 - should be set to NULL</font><br>
<a name="monodelphis_domestica_core_39_3:DisplayXref"></a> <font color='red'><strong>DisplayXref: </strong>33804 transcripts (out of a total of 33914) have display_xref_ids pointing to non-existent xrefs.</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>gene_attrib has zero rows</font><br>
<a name="monodelphis_domestica_core_39_3:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>karyotype has zero rows</font><br>
<a name="monodelphis_domestica_core_39_3:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>regulatory_search_region has zero rows</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 17577 length 2048225</font><br>
<a name="monodelphis_domestica_core_39_3:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 17580 length 2048225</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>Only 104 genes have KNOWN status in the current database, compared with 16010 in the previous database</font><br>
<a name="monodelphis_domestica_core_39_3:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>Only 110 transcripts have KNOWN status in the current database, compared with 964 in the previous database</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_rRNA is not a valid gene type name.</font><br>
<a name="monodelphis_domestica_core_39_3:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_tRNA is not a valid gene type name.</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:Karyotype"></a> <font color='red'><strong>Karyotype: </strong>Karyotype table is empty</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:MappingSession"></a> <font color='red'><strong>MappingSession: </strong>ALL/LATEST mapping session does not seem to have the most stable_id_event entries</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:OligoProbes2Genome"></a> <font color='grey'><strong>OligoProbes2Genome: </strong>No rows in oligo_feature, so not checking for entry in meta_coord</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:ProteinFeatureTranslation"></a> <font color='red'><strong>ProteinFeatureTranslation: </strong>protein_feature table has 496 features that are longer than the translation</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:ProteinFeatures"></a> <font color='red'><strong>ProteinFeatures: </strong>83 protein features have seq_start < 0</font><br>
<a name="monodelphis_domestica_core_39_3:ProteinFeatures"></a> <font color='red'><strong>ProteinFeatures: </strong>16 protein features have hit_start < 0</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>monodelphis_domestica_core_38_2b has 2618326 rows in supporting_feature but monodelphis_domestica_core_39_3 only has 1639043</font><br>
<a name="monodelphis_domestica_core_39_3:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>monodelphis_domestica_core_38_2b has 5968 rows in seq_region_attrib but monodelphis_domestica_core_39_3 only has 92</font><br>
<a name="monodelphis_domestica_core_39_3:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>monodelphis_domestica_core_38_2b has 2469 rows in transcript_attrib but monodelphis_domestica_core_39_3 only has 1071</font><br>
</p><p>
<a name="monodelphis_domestica_core_39_3:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table pmatch_feature exists in monodelphis_domestica_core_39_3 but not in master_schema_39</font><br>
<a name="monodelphis_domestica_core_39_3:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table protein exists in monodelphis_domestica_core_39_3 but not in master_schema_39</font><br>
<a name="monodelphis_domestica_core_39_3:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table name discrepancy detected, skipping rest of checks</font><br>
</p>
<hr>
<h2>Detailed reports by test case</h2>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.BigGeneExon">org.ensembl.healthcheck.testcase.generic.BigGeneExon</a></h3>
<p>
<a name="BigGeneExon:monodelphis_domestica_core_39_3"></a> <font color='grey'><strong>monodelphis_domestica_core_39_3: </strong>18 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes">org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes</a></h3>
<p>
<a name="ComparePreviousVersionBiotypes:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>monodelphis_domestica_core_38_2b has 2 biotypes of type Mt-rRNA but monodelphis_domestica_core_39_3 has none</font><br>
<a name="ComparePreviousVersionBiotypes:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>monodelphis_domestica_core_38_2b has 21 biotypes of type Mt-tRNA but monodelphis_domestica_core_39_3 has none</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows">org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows</a></h3>
<p>
<a name="ComparePreviousVersionTableRows:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>monodelphis_domestica_core_38_2b has 2618326 rows in supporting_feature but monodelphis_domestica_core_39_3 only has 1639043</font><br>
<a name="ComparePreviousVersionTableRows:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>monodelphis_domestica_core_38_2b has 5968 rows in seq_region_attrib but monodelphis_domestica_core_39_3 only has 92</font><br>
<a name="ComparePreviousVersionTableRows:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>monodelphis_domestica_core_38_2b has 2469 rows in transcript_attrib but monodelphis_domestica_core_39_3 only has 1071</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.CompareSchema">org.ensembl.healthcheck.testcase.generic.CompareSchema</a></h3>
<p>
<a name="CompareSchema:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Table pmatch_feature exists in monodelphis_domestica_core_39_3 but not in master_schema_39</font><br>
<a name="CompareSchema:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Table protein exists in monodelphis_domestica_core_39_3 but not in master_schema_39</font><br>
<a name="CompareSchema:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Table name discrepancy detected, skipping rest of checks</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DisplayXref">org.ensembl.healthcheck.testcase.generic.DisplayXref</a></h3>
<p>
<a name="DisplayXref:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>7513 gene display_xrefs are set to 0 - should be set to NULL</font><br>
<a name="DisplayXref:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>7513 genes (out of a total of 20907) have display_xref_ids pointing to non-existent xrefs.</font><br>
<a name="DisplayXref:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>33804 transcript display_xrefs are set to 0 - should be set to NULL</font><br>
<a name="DisplayXref:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>33804 transcripts (out of a total of 33914) have display_xref_ids pointing to non-existent xrefs.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.EmptyTables">org.ensembl.healthcheck.testcase.generic.EmptyTables</a></h3>
<p>
<a name="EmptyTables:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>gene_attrib has zero rows</font><br>
<a name="EmptyTables:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>karyotype has zero rows</font><br>
<a name="EmptyTables:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>regulatory_search_region has zero rows</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ExonStrandOrder">org.ensembl.healthcheck.testcase.generic.ExonStrandOrder</a></h3>
<p>
<a name="ExonStrandOrder:monodelphis_domestica_core_39_3"></a> <font color='black'><strong>monodelphis_domestica_core_39_3: </strong>Long transcript 17577 length 2048225</font><br>
<a name="ExonStrandOrder:monodelphis_domestica_core_39_3"></a> <font color='black'><strong>monodelphis_domestica_core_39_3: </strong>Long transcript 17580 length 2048225</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.GeneStatus">org.ensembl.healthcheck.testcase.generic.GeneStatus</a></h3>
<p>
<a name="GeneStatus:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Only 104 genes have KNOWN status in the current database, compared with 16010 in the previous database</font><br>
<a name="GeneStatus:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Only 110 transcripts have KNOWN status in the current database, compared with 964 in the previous database</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.GeneType">org.ensembl.healthcheck.testcase.generic.GeneType</a></h3>
<p>
<a name="GeneType:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Mt_rRNA is not a valid gene type name.</font><br>
<a name="GeneType:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Mt_tRNA is not a valid gene type name.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.Karyotype">org.ensembl.healthcheck.testcase.generic.Karyotype</a></h3>
<p>
<a name="Karyotype:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>Karyotype table is empty</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.MappingSession">org.ensembl.healthcheck.testcase.generic.MappingSession</a></h3>
<p>
<a name="MappingSession:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>ALL/LATEST mapping session does not seem to have the most stable_id_event entries</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.OligoProbes2Genome">org.ensembl.healthcheck.testcase.generic.OligoProbes2Genome</a></h3>
<p>
<a name="OligoProbes2Genome:monodelphis_domestica_core_39_3"></a> <font color='grey'><strong>monodelphis_domestica_core_39_3: </strong>No rows in oligo_feature, so not checking for entry in meta_coord</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation">org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation</a></h3>
<p>
<a name="ProteinFeatureTranslation:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>protein_feature table has 496 features that are longer than the translation</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ProteinFeatures">org.ensembl.healthcheck.testcase.generic.ProteinFeatures</a></h3>
<p>
<a name="ProteinFeatures:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>83 protein features have seq_start < 0</font><br>
<a name="ProteinFeatures:monodelphis_domestica_core_39_3"></a> <font color='red'><strong>monodelphis_domestica_core_39_3: </strong>16 protein features have hit_start < 0</font><br>
</p>
<hr>
<p>Test run was started at Wed May 31 03:48:00 BST 2006 and finished at Wed May 31 05:01:39 BST 2006<br>
 Run time 1 hours 13 min 38s </p>
<h4>Configuration used:</h4>
<pre>
Tests/groups run:   release<br>
Database host:      ecs2:3364<br>
Database names:     ^monodelphis_domestica_(core|otherfeatures|cdna|vega)_[0-9]+_.*<br>
Output file:        web_monodelphis_domestica.html<br>
Output level:       info<br>
</pre>
</div>
</body>
</html>
<hr>
