<html>
<head>
<style type="text/css" media="all">
@import url(http://www.ensembl.org/css/ensembl.css);
@import url(http://www.ensembl.org/css/content.css);
#page ul li { list-style-type:none; list-style-image: none; margin-left: -2em }
</style>
<title>Healthcheck results for rattus norvegicus databases</title>
</head>
<body>
<div id='page'><div id='i1'><div id='i2'><div class='sptop'>&nbsp;</div>
<div id='release'>Healthcheck results for rattus norvegicus databases</div>
<hr>
<div id='related'><div id='related-box'>
<h2>Results by database</h2>
<ul>
<li><a href="#master_schema_39">master_schema_39</a></li>
<li><a href="#rattus_norvegicus_core_39_34i">r_norvegicus_core_39_34i</a></li>
<li><a href="#rattus_norvegicus_otherfeatures_39_34i">r_norvegicus_otherfeatures_39_34i</a></li>
</ul>
<h2>Results by test</h2>
<ul>
<li><a href="#Accession">Accession</a></li>
<li><a href="#AnalysisDescription">AnalysisDescription</a></li>
<li><a href="#AssemblyException">AssemblyException</a></li>
<li><a href="#AssemblyMultipleOverlap">AssemblyMultipleOverlap</a></li>
<li><a href="#AssemblySeqregion">AssemblySeqregion</a></li>
<li><a href="#AssemblyTablesAcrossSpecies">AssemblyTablesAcrossSpecies</a></li>
<li><a href="#AttribTypeAcrossSpecies">AttribTypeAcrossSpecies</a></li>
<li><a href="#BigGeneExon">BigGeneExon</a></li>
<li><a href="#Biotypes">Biotypes</a></li>
<li><a href="#BlankInfoType">BlankInfoType</a></li>
<li><a href="#ComparePreviousVersionBiotypes">ComparePreviousVersionBiotyp</a></li>
<li><a href="#ComparePreviousVersionTableRows">ComparePreviousVersionTableR</a></li>
<li><a href="#ComparePreviousVersionXrefs">ComparePreviousVersionXrefs</a></li>
<li><a href="#CompareSchema">CompareSchema</a></li>
<li><a href="#CoordSystemAcrossSpecies">CoordSystemAcrossSpecies</a></li>
<li><a href="#CoreForeignKeys">CoreForeignKeys</a></li>
<li><a href="#DNAEmpty">DNAEmpty</a></li>
<li><a href="#DNATablesAcrossSpecies">DNATablesAcrossSpecies</a></li>
<li><a href="#DensityFeatures">DensityFeatures</a></li>
<li><a href="#DisplayLabels">DisplayLabels</a></li>
<li><a href="#DisplayMarkerSynonyms">DisplayMarkerSynonyms</a></li>
<li><a href="#DisplayXref">DisplayXref</a></li>
<li><a href="#DuplicateAssembly">DuplicateAssembly</a></li>
<li><a href="#ESTStableID">ESTStableID</a></li>
<li><a href="#ESTStableIDs">ESTStableIDs</a></li>
<li><a href="#EmptyTables">EmptyTables</a></li>
<li><a href="#ExonStrandOrder">ExonStrandOrder</a></li>
<li><a href="#ExonTranscriptStartEnd">ExonTranscriptStartEnd</a></li>
<li><a href="#ExternalDBDisplayName">ExternalDBDisplayName</a></li>
<li><a href="#FeatureAnalysis">FeatureAnalysis</a></li>
<li><a href="#FeatureCoords">FeatureCoords</a></li>
<li><a href="#GOXrefs">GOXrefs</a></li>
<li><a href="#GeneStatus">GeneStatus</a></li>
<li><a href="#GeneType">GeneType</a></li>
<li><a href="#InterproDescriptions">InterproDescriptions</a></li>
<li><a href="#Karyotype">Karyotype</a></li>
<li><a href="#LogicNamesDisplayable">LogicNamesDisplayable</a></li>
<li><a href="#MappingSession">MappingSession</a></li>
<li><a href="#MarkerFeatures">MarkerFeatures</a></li>
<li><a href="#Meta">Meta</a></li>
<li><a href="#MetaCoord">MetaCoord</a></li>
<li><a href="#MetaCrossSpecies">MetaCrossSpecies</a></li>
<li><a href="#OligoMismatches">OligoMismatches</a></li>
<li><a href="#OligoXrefs">OligoXrefs</a></li>
<li><a href="#PredictedXrefs">PredictedXrefs</a></li>
<li><a href="#ProteinFeatureTranslation">ProteinFeatureTranslation</a></li>
<li><a href="#ProteinFeatures">ProteinFeatures</a></li>
<li><a href="#ProteinIDXrefsNoVersion">ProteinIDXrefsNoVersion</a></li>
<li><a href="#Pseudogene">Pseudogene</a></li>
<li><a href="#RepeatConsensus">RepeatConsensus</a></li>
<li><a href="#RepeatFeature">RepeatFeature</a></li>
<li><a href="#SeqRegionAcrossSpecies">SeqRegionAcrossSpecies</a></li>
<li><a href="#SeqRegionAttribAcrossSpecies">SeqRegionAttribAcrossSpecies</a></li>
<li><a href="#SeqRegionCoordSystem">SeqRegionCoordSystem</a></li>
<li><a href="#SeqRegionsTopLevel">SeqRegionsTopLevel</a></li>
<li><a href="#SingleDBCollations">SingleDBCollations</a></li>
<li><a href="#StableID">StableID</a></li>
<li><a href="#Strand">Strand</a></li>
<li><a href="#TranscriptsTranslate">TranscriptsTranslate</a></li>
<li><a href="#TranslationStartEnd">TranslationStartEnd</a></li>
<li><a href="#TranslationStartEndExon">TranslationStartEndExon</a></li>
<li><a href="#XrefTypes">XrefTypes</a></li>
</ul>
</div></div>
<h2>Summary</h2>
<p><strong><font color='green' size=-1>71</font> tests passed and <font color='red' size=-1>17</font> failed out of a total of 88 tests run.</strong></p>
<hr>
<h2>Summary by database</h2>
<p><table class='ss'>
<tr><th>Database</th><th>Passed</th><th>Failed</th></tr>
<tr><td><a href="#master_schema_39"><font color='red' size=-1>master_schema_39</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#rattus_norvegicus_core_39_34i"><font color='red' size=-1>rattus_norvegicus_core_39_34i</font></a></td><td><font color='green' size=-1>42</font></td><td><font color='red' size=-1>13</font></td></tr>
<tr><td><a href="#rattus_norvegicus_otherfeatures_39_34i"><font color='red' size=-1>rattus_norvegicus_otherfeatures_39_34i</font></a></td><td><font color='green' size=-1>29</font></td><td><font color='red' size=-1>3</font></td></tr>
</table></p>
<hr>
<h2>Summary by test</h2>
<p><table class='ss'>
<tr><th>Test</th><th>Passed</th><th>Failed</th></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Accession"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Accession</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AnalysisDescription"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.AnalysisDescription</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyException"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblyException</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyMultipleOverlap"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblyMultipleOverlap</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblySeqregion"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblySeqregion</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.BigGeneExon"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.BigGeneExon</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Biotypes"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Biotypes</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.BlankInfoType"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.BlankInfoType</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CompareSchema"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.CompareSchema</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CoordSystemAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.CoordSystemAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CoreForeignKeys"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.CoreForeignKeys</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DNAEmpty"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DNAEmpty</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DensityFeatures"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DensityFeatures</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayLabels"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DisplayLabels</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayXref"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.DisplayXref</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateAssembly"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.DuplicateAssembly</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ESTStableID"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ESTStableID</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ESTStableIDs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ESTStableIDs</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.EmptyTables"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.EmptyTables</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExonStrandOrder"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ExonStrandOrder</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.FeatureAnalysis"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.FeatureAnalysis</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.FeatureCoords"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.FeatureCoords</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GOXrefs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.GOXrefs</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GeneStatus"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.GeneStatus</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GeneType"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.GeneType</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.InterproDescriptions"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.InterproDescriptions</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Karyotype"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Karyotype</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MappingSession"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.MappingSession</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MarkerFeatures"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.MarkerFeatures</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Meta"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Meta</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MetaCoord"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.MetaCoord</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.OligoMismatches"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.OligoMismatches</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.OligoXrefs"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.OligoXrefs</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.PredictedXrefs"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.PredictedXrefs</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatures"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ProteinFeatures</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Pseudogene"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Pseudogene</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.RepeatConsensus"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.RepeatConsensus</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.RepeatFeature"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.RepeatFeature</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionAcrossSpecies"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionAcrossSpecies</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies</font></a></td><td><font color='green' size=-1>0</font></td><td><font color='red' size=-1>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SingleDBCollations"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.SingleDBCollations</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.StableID"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.StableID</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Strand"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.Strand</font></a></td><td><font color='green' size=-1>2</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEnd"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.TranslationStartEnd</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon"><font color='green' size=-1>org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.XrefTypes"><font color='red' size=-1>org.ensembl.healthcheck.testcase.generic.XrefTypes</font></a></td><td><font color='green' size=-1>1</font></td><td><font color='red' size=-1>1</font></td></tr>
</table></p>
<hr>
<h2>Detailed reports by database</h2>
<h3 class='boxed'><a name="master_schema_39">master_schema_39</a></h3>
<p>
<a name="master_schema_39:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>master_schema_39 and rattus_norvegicus_core_39_34i oligo_feature differ at row 8 column 2 (Type) Values: int(10) unsigned, int(11)</font><br>
<a name="master_schema_39:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>master_schema_39 and rattus_norvegicus_otherfeatures_39_34i oligo_feature differ at row 8 column 2 (Type) Values: int(10) unsigned, int(11)</font><br>
</p>
<h3 class='boxed'><a name="rattus_norvegicus_core_39_34i">rattus_norvegicus_core_39_34i</a></h3>
<p>
<a name="rattus_norvegicus_core_39_34i:BigGeneExon"></a> <font color='grey'><strong>BigGeneExon: </strong>23 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:CoreForeignKeys"></a> <font color='grey'><strong>CoreForeignKeys: </strong>identity_xref.object_xref_id 2672839 is not linked.</font><br>
<a name="rattus_norvegicus_core_39_34i:CoreForeignKeys"></a> <font color='red'><strong>CoreForeignKeys: </strong>identity_xref <-> object_xref has 1 unlinked entries</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:DisplayXref"></a> <font color='red'><strong>DisplayXref: </strong>1 genes (out of a total of 25489) have display_xref_ids pointing to non-existent xrefs.</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>gene_attrib has zero rows</font><br>
<a name="rattus_norvegicus_core_39_34i:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>regulatory_search_region has zero rows</font><br>
<a name="rattus_norvegicus_core_39_34i:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>transcript_supporting_feature has zero rows</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 12520 length 2295759</font><br>
<a name="rattus_norvegicus_core_39_34i:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 12591 length 2619506</font><br>
<a name="rattus_norvegicus_core_39_34i:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 16188 length 2318978</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 5108, logic name miRNA is not used in any feature table</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>Only 11353 genes have KNOWN status in the current database, compared with 18063 in the previous database</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_rRNA is not a valid gene type name.</font><br>
<a name="rattus_norvegicus_core_39_34i:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_tRNA is not a valid gene type name.</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:MappingSession"></a> <font color='red'><strong>MappingSession: </strong>Old/new names rattus_norvegicus_core_29_3f rattus_norvegicus_core_20_3 do not chain properly</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:OligoXrefs"></a> <font color='red'><strong>OligoXrefs: </strong>Chromosome MT( seq_region_id=254355) has no associated Oligo xrefs.</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:XrefTypes"></a> <font color='red'><strong>XrefTypes: </strong>External DB with ID 3300 (miRNA_Registry) is associated with Gene as well as Transcript</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:AssemblyTablesAcrossSpecies"></a> <font color='red'><strong>AssemblyTablesAcrossSpecies: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 127180, 127179</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:AttribTypeAcrossSpecies"></a> <font color='red'><strong>AttribTypeAcrossSpecies: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 77, 51</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:SeqRegionAttribAcrossSpecies"></a> <font color='red'><strong>SeqRegionAttribAcrossSpecies: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 218, 21</font><br>
</p><p>
<a name="rattus_norvegicus_core_39_34i:SeqRegionAcrossSpecies"></a> <font color='red'><strong>SeqRegionAcrossSpecies: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 127179, 127177</font><br>
</p>
<h3 class='boxed'><a name="rattus_norvegicus_otherfeatures_39_34i">rattus_norvegicus_otherfeatures_39_34i</a></h3>
<p>
<a name="rattus_norvegicus_otherfeatures_39_34i:AnalysisDescription"></a> <font color='red'><strong>AnalysisDescription: </strong>Analysis genomewise is used in gene but has no entry in analysis_description</font><br>
<a name="rattus_norvegicus_otherfeatures_39_34i:AnalysisDescription"></a> <font color='red'><strong>AnalysisDescription: </strong>Analysis genomewise is used in transcript but has no entry in analysis_description</font><br>
</p><p>
<a name="rattus_norvegicus_otherfeatures_39_34i:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>No genes have status KNOWN</font><br>
</p><p>
<a name="rattus_norvegicus_otherfeatures_39_34i:GeneType"></a> <font color='red'><strong>GeneType: </strong>genomewise (protein_coding) is not a valid analysis logic name.</font><br>
</p>
<hr>
<h2>Detailed reports by test case</h2>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.AnalysisDescription">org.ensembl.healthcheck.testcase.generic.AnalysisDescription</a></h3>
<p>
<a name="AnalysisDescription:rattus_norvegicus_otherfeatures_39_34i"></a> <font color='red'><strong>rattus_norvegicus_otherfeatures_39_34i: </strong>Analysis genomewise is used in gene but has no entry in analysis_description</font><br>
<a name="AnalysisDescription:rattus_norvegicus_otherfeatures_39_34i"></a> <font color='red'><strong>rattus_norvegicus_otherfeatures_39_34i: </strong>Analysis genomewise is used in transcript but has no entry in analysis_description</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies">org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies</a></h3>
<p>
<a name="AssemblyTablesAcrossSpecies:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 127180, 127179</font><br>
</p><p>
<a name="AssemblyTablesAcrossSpecies:"></a> <font color='red'><strong>: </strong>assembly tables different</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies">org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies</a></h3>
<p>
<a name="AttribTypeAcrossSpecies:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 77, 51</font><br>
</p><p>
<a name="AttribTypeAcrossSpecies:"></a> <font color='red'><strong>: </strong>attrib_type tables are different</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.BigGeneExon">org.ensembl.healthcheck.testcase.generic.BigGeneExon</a></h3>
<p>
<a name="BigGeneExon:rattus_norvegicus_core_39_34i"></a> <font color='grey'><strong>rattus_norvegicus_core_39_34i: </strong>23 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.CompareSchema">org.ensembl.healthcheck.testcase.generic.CompareSchema</a></h3>
<p>
<a name="CompareSchema:master_schema_39"></a> <font color='red'><strong>master_schema_39: </strong>master_schema_39 and rattus_norvegicus_core_39_34i oligo_feature differ at row 8 column 2 (Type) Values: int(10) unsigned, int(11)</font><br>
<a name="CompareSchema:master_schema_39"></a> <font color='red'><strong>master_schema_39: </strong>master_schema_39 and rattus_norvegicus_otherfeatures_39_34i oligo_feature differ at row 8 column 2 (Type) Values: int(10) unsigned, int(11)</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.CoreForeignKeys">org.ensembl.healthcheck.testcase.generic.CoreForeignKeys</a></h3>
<p>
<a name="CoreForeignKeys:rattus_norvegicus_core_39_34i"></a> <font color='grey'><strong>rattus_norvegicus_core_39_34i: </strong>identity_xref.object_xref_id 2672839 is not linked.</font><br>
<a name="CoreForeignKeys:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>identity_xref <-> object_xref has 1 unlinked entries</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DisplayXref">org.ensembl.healthcheck.testcase.generic.DisplayXref</a></h3>
<p>
<a name="DisplayXref:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>1 genes (out of a total of 25489) have display_xref_ids pointing to non-existent xrefs.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.EmptyTables">org.ensembl.healthcheck.testcase.generic.EmptyTables</a></h3>
<p>
<a name="EmptyTables:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>gene_attrib has zero rows</font><br>
<a name="EmptyTables:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>regulatory_search_region has zero rows</font><br>
<a name="EmptyTables:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>transcript_supporting_feature has zero rows</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ExonStrandOrder">org.ensembl.healthcheck.testcase.generic.ExonStrandOrder</a></h3>
<p>
<a name="ExonStrandOrder:rattus_norvegicus_core_39_34i"></a> <font color='black'><strong>rattus_norvegicus_core_39_34i: </strong>Long transcript 12520 length 2295759</font><br>
<a name="ExonStrandOrder:rattus_norvegicus_core_39_34i"></a> <font color='black'><strong>rattus_norvegicus_core_39_34i: </strong>Long transcript 12591 length 2619506</font><br>
<a name="ExonStrandOrder:rattus_norvegicus_core_39_34i"></a> <font color='black'><strong>rattus_norvegicus_core_39_34i: </strong>Long transcript 16188 length 2318978</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.FeatureAnalysis">org.ensembl.healthcheck.testcase.generic.FeatureAnalysis</a></h3>
<p>
<a name="FeatureAnalysis:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>Analysis with ID 5108, logic name miRNA is not used in any feature table</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.GeneStatus">org.ensembl.healthcheck.testcase.generic.GeneStatus</a></h3>
<p>
<a name="GeneStatus:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>Only 11353 genes have KNOWN status in the current database, compared with 18063 in the previous database</font><br>
</p><p>
<a name="GeneStatus:rattus_norvegicus_otherfeatures_39_34i"></a> <font color='red'><strong>rattus_norvegicus_otherfeatures_39_34i: </strong>No genes have status KNOWN</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.GeneType">org.ensembl.healthcheck.testcase.generic.GeneType</a></h3>
<p>
<a name="GeneType:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>Mt_rRNA is not a valid gene type name.</font><br>
<a name="GeneType:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>Mt_tRNA is not a valid gene type name.</font><br>
</p><p>
<a name="GeneType:rattus_norvegicus_otherfeatures_39_34i"></a> <font color='red'><strong>rattus_norvegicus_otherfeatures_39_34i: </strong>genomewise (protein_coding) is not a valid analysis logic name.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.MappingSession">org.ensembl.healthcheck.testcase.generic.MappingSession</a></h3>
<p>
<a name="MappingSession:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>Old/new names rattus_norvegicus_core_29_3f rattus_norvegicus_core_20_3 do not chain properly</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.OligoXrefs">org.ensembl.healthcheck.testcase.generic.OligoXrefs</a></h3>
<p>
<a name="OligoXrefs:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>Chromosome MT( seq_region_id=254355) has no associated Oligo xrefs.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.SeqRegionAcrossSpecies">org.ensembl.healthcheck.testcase.generic.SeqRegionAcrossSpecies</a></h3>
<p>
<a name="SeqRegionAcrossSpecies:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 127179, 127177</font><br>
</p><p>
<a name="SeqRegionAcrossSpecies:"></a> <font color='red'><strong>: </strong>seq_region tables are different</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies">org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies</a></h3>
<p>
<a name="SeqRegionAttribAcrossSpecies:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>rattus_norvegicus_core_39_34i and rattus_norvegicus_otherfeatures_39_34i  differ at row 1 column 1 (COUNT(*)) Values: 218, 21</font><br>
</p><p>
<a name="SeqRegionAttribAcrossSpecies:"></a> <font color='red'><strong>: </strong>seq_region_attribs are different</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.XrefTypes">org.ensembl.healthcheck.testcase.generic.XrefTypes</a></h3>
<p>
<a name="XrefTypes:rattus_norvegicus_core_39_34i"></a> <font color='red'><strong>rattus_norvegicus_core_39_34i: </strong>External DB with ID 3300 (miRNA_Registry) is associated with Gene as well as Transcript</font><br>
</p>
<hr>
<p>Test run was started at Wed May 31 06:45:05 BST 2006 and finished at Wed May 31 07:00:43 BST 2006<br>
 Run time 15 min 37s </p>
<h4>Configuration used:</h4>
<pre>
Tests/groups run:   release<br>
Database host:      ecs2:3364<br>
Database names:     ^rattus_norvegicus_(core|otherfeatures|cdna|vega)_[0-9]+_.*<br>
Output file:        web_rattus_norvegicus.html<br>
Output level:       info<br>
</pre>
</div>
</body>
</html>
<hr>
