<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<HTML>
  <HEAD>
    <TITLE>Ensj Heathcheck tests</TITLE>
  </HEAD>
  <BODY>
    
  <h1>Ensj Heathcheck tests</h1>
    
<p>All tests:</p>

<table border='1' cellpadding='5'>

<tr><th bgcolor="#eeeeee">Test Name</th><th bgcolor="#eeeeee">Group(s)</th><th bgcolor="#eeeeee">Description</th></tr>

<tr><td class="lfooter"><strong>UnderscoreInAssemblyTypeTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>Checks for the presence of _ characters in assembly.type</font></td></tr>
<tr><td class="lfooter"><strong>CheckTranscriptsTranslateTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that all transcripts have an associated translation</font></td></tr>
<tr><td class="lfooter"><strong>CheckAllExonsBelongToTranscriptTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that all exons belong to a transcript.</font></td></tr>
<tr><td class="lfooter"><strong>BlankNonNullExternalDBTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>Checks for columns in the external_db table that are marked as NON NULL but contain null or blank values</font></td></tr>
<tr><td class="lfooter"><strong>CheckAllTranscriptsBelongToAGeneTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that all transcripts belong to a gene.</font></td></tr>
<tr><td class="lfooter"><strong>CheckDuplicateExonsTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check for duplicate exons.</font></td></tr>
<tr><td class="lfooter"><strong>CheckMetaDataTableTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>Check that the meta table exists, has data, the entries correspond to the database name, and that the values in assembly.type match what's in the meta table</font></td></tr>
<tr><td class="lfooter"><strong>ExternalDBDescribeTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>Check that the external_db table is the same in all databases.</font></td></tr>
<tr><td class="lfooter"><strong>CheckAllGenesHaveTranscriptsTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that all genes have associated transcripts</font></td></tr>
<tr><td class="lfooter"><strong>CheckDNATablesTestCase</strong></td><td>pre_mart<br></td><td><font size=-1>Check that the DNA table exists, and all DBs for a particular species have the same entries.</font></td></tr>
<tr><td class="lfooter"><strong>CheckAllTranscriptsHaveExonsTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that all transcripts have an associated exon.</font></td></tr>
<tr><td class="lfooter"><strong>CheckAssemblyTablesTestCase</strong></td><td>pre_mart<br></td><td><font size=-1>Check that the assembly table is present, and contains the same information for each species.</font></td></tr>
<tr><td class="lfooter"><strong>CheckTranslationStartEndTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that if the start and end of translation is on the same exon, that start < end.</font></td></tr>
<tr><td class="lfooter"><strong>CheckCoreTablesHaveDataTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>Verify that important tables in core DBs actually have data.</font></td></tr>
<tr><td class="lfooter"><strong>CoreForeignKeyTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>Check for broken foreign-key realtionships.</font></td></tr>
<tr><td class="lfooter"><strong>CheckExonCoordsTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that exon co-ordinates are sensible.</font></td></tr>
<tr><td class="lfooter"><strong>EnsjTrialTestCase</strong></td><td>ensj<br></td><td><font size=-1>An example of how to integrate ensj-core functionality. Checks that all genes on a chromosome have transcripts. For illustration only - not intended to be used in production.</font></td></tr>
<tr><td class="lfooter"><strong>CheckFeatureCoordsTestCase</strong></td><td>post_genebuild<br></td><td><font size=-1>Check that feature co-ords (DNA and protein) make sense.</font></td></tr>
<tr><td class="lfooter"><strong>ExonStrandOrderTestCase</strong></td><td>db_constraints<br></td><td><font size=-1>No description set for this test.</font></td></tr>
<tr><td class="lfooter"><strong>GlennsRowCountTestCase</strong></td><td>dev<br></td><td><font size=-1>No description set for this test.</font></td></tr>


</table>

<p>Tests by group:</p>

<p><strong>db_constraints</strong></p>UnderscoreInAssemblyTypeTestCase<br>BlankNonNullExternalDBTestCase<br>CheckMetaDataTableTestCase<br>ExternalDBDescribeTestCase<br>CheckCoreTablesHaveDataTestCase<br>CoreForeignKeyTestCase<br>ExonStrandOrderTestCase<br><p><strong>post_genebuild</strong></p>CheckTranscriptsTranslateTestCase<br>CheckAllExonsBelongToTranscriptTestCase<br>CheckAllTranscriptsBelongToAGeneTestCase<br>CheckDuplicateExonsTestCase<br>CheckAllGenesHaveTranscriptsTestCase<br>CheckAllTranscriptsHaveExonsTestCase<br>CheckTranslationStartEndTestCase<br>CheckExonCoordsTestCase<br>CheckFeatureCoordsTestCase<br><p><strong>pre_mart</strong></p>CheckDNATablesTestCase<br>CheckAssemblyTablesTestCase<br><p><strong>ensj</strong></p>EnsjTrialTestCase<br><p><strong>dev</strong></p>GlennsRowCountTestCase<br>

<p> Note that every test is a member of the group "all" by default.<p>

<hr noshade size='1'>

<p>Last updated:
Wed Apr 23 12:17:21 BST 2003
</p>

  </BODY>
</HTML>
