<html>
<head>
<style type="text/css" media="all">
@import url(http://www.ensembl.org/css/ensembl.css);
@import url(http://www.ensembl.org/css/content.css);
#page ul li { list-style-type:none; list-style-image: none; margin-left: -2em }
</style>
<title>Healthcheck results for homo sapiens databases</title>
</head>
<body>
<div id='page'><div id='i1'><div id='i2'><div class='sptop'>&nbsp;</div>
<div id='release'>Healthcheck results for homo sapiens databases</div>
<hr>
<div id='related'><div id='related-box'>
<h2>Results by database</h2>
<ul>
<li><a href="#homo_sapiens_cdna_38_36">homo_sapiens_cdna_38_36</a></li>
<li><a href="#homo_sapiens_core_38_36">homo_sapiens_core_38_36</a></li>
<li><a href="#homo_sapiens_otherfeatures_38_36">homo_sapiens_otherfeatures_38_36</a></li>
<li><a href="#homo_sapiens_vega_38_36">homo_sapiens_vega_38_36</a></li>
<li><a href="#master_schema_38">master_schema_38</a></li>
</ul>
<h2>Results by test</h2>
<ul>
<li><a href="#org.ensembl.healthcheck.testcase.generic.Accession">Accession</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.AnalysisDescription">AnalysisDescription</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyException">AssemblyException</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.AssemblySeqregion">AssemblySeqregion</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies">AssemblyTablesAcrossSpecies</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies">AttribTypeAcrossSpecies</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.BigGeneExon">BigGeneExon</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.Biotypes">Biotypes</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.BlankInfoType">BlankInfoType</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes">ComparePreviousVersionBiotypes</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows">ComparePreviousVersionTableRows</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs">ComparePreviousVersionXrefs</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.CompareSchema">CompareSchema</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.CoreForeignKeys">CoreForeignKeys</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DNAEmpty">DNAEmpty</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies">DNATablesAcrossSpecies</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DensityFeatures">DensityFeatures</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DisplayLabels">DisplayLabels</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms">DisplayMarkerSynonyms</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DisplayXref">DisplayXref</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateAssembly">DuplicateAssembly</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateExons">DuplicateExons</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateGenes">DuplicateGenes</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ESTStableID">ESTStableID</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ESTStableIDs">ESTStableIDs</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.EmptyTables">EmptyTables</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ExonStrandOrder">ExonStrandOrder</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd">ExonTranscriptStartEnd</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName">ExternalDBDisplayName</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.FeatureAnalysis">FeatureAnalysis</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.FeatureCoords">FeatureCoords</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.GOXrefs">GOXrefs</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.GeneStatus">GeneStatus</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.GeneType">GeneType</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.HumanCCDS">HumanCCDS</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.InterproDescriptions">InterproDescriptions</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.Karyotype">Karyotype</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable">LogicNamesDisplayable</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.MappingSession">MappingSession</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.MarkerFeatures">MarkerFeatures</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.Meta">Meta</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.MetaCoord">MetaCoord</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies">MetaCrossSpecies</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.OligoMismatches">OligoMismatches</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.OligoXrefs">OligoXrefs</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.PredictedXrefs">PredictedXrefs</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation">ProteinFeatureTranslation</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatures">ProteinFeatures</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion">ProteinIDXrefsNoVersion</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.Pseudogene">Pseudogene</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.RepeatConsensus">RepeatConsensus</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.RepeatFeature">RepeatFeature</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies">SeqRegionAttribAcrossSpecies</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem">SeqRegionCoordSystem</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionName">SeqRegionName</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel">SeqRegionsTopLevel</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.SingleDBCollations">SingleDBCollations</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.StableID">StableID</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.Strand">Strand</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate">TranscriptsTranslate</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEnd">TranslationStartEnd</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon">TranslationStartEndExon</a></li>
<li><a href="#org.ensembl.healthcheck.testcase.generic.XrefTypes">XrefTypes</a></li>
</ul>
</div></div>
<h2>Summary</h2>
<p><strong><font color='green'>133</font> tests passed and <font color='red'>41</font> failed out of a total of 174 tests run.</strong></p>
<hr>
<h2>Summary by database</h2>
<p><table class='ss'>
<tr><th>Database</th><th>Passed</th><th>Failed</th></tr>
<tr><td><a href="#homo_sapiens_cdna_38_36"><font color='red'>homo_sapiens_cdna_38_36</font></a></td><td><font color='green'>35</font></td><td><font color='red'>3</font></td></tr>
<tr><td><a href="#homo_sapiens_core_38_36"><font color='red'>homo_sapiens_core_38_36</font></a></td><td><font color='green'>39</font></td><td><font color='red'>20</font></td></tr>
<tr><td><a href="#homo_sapiens_otherfeatures_38_36"><font color='red'>homo_sapiens_otherfeatures_38_36</font></a></td><td><font color='green'>27</font></td><td><font color='red'>5</font></td></tr>
<tr><td><a href="#homo_sapiens_vega_38_36"><font color='red'>homo_sapiens_vega_38_36</font></a></td><td><font color='green'>32</font></td><td><font color='red'>12</font></td></tr>
<tr><td><a href="#master_schema_38"><font color='red'>master_schema_38</font></a></td><td><font color='green'>0</font></td><td><font color='red'>1</font></td></tr>
</table></p>
<hr>
<h2>Summary by test</h2>
<p><table class='ss'>
<tr><th>Test</th><th>Passed</th><th>Failed</th></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Accession"><font color='green'>org.ensembl.healthcheck.testcase.generic.Accession</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AnalysisDescription"><font color='green'>org.ensembl.healthcheck.testcase.generic.AnalysisDescription</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyException"><font color='green'>org.ensembl.healthcheck.testcase.generic.AssemblyException</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblySeqregion"><font color='green'>org.ensembl.healthcheck.testcase.generic.AssemblySeqregion</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies"><font color='red'>org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies"><font color='red'>org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.BigGeneExon"><font color='red'>org.ensembl.healthcheck.testcase.generic.BigGeneExon</font></a></td><td><font color='green'>1</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Biotypes"><font color='green'>org.ensembl.healthcheck.testcase.generic.Biotypes</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.BlankInfoType"><font color='green'>org.ensembl.healthcheck.testcase.generic.BlankInfoType</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes"><font color='green'>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionBiotypes</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows"><font color='red'>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows</font></a></td><td><font color='green'>3</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs"><font color='green'>org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionXrefs</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CompareSchema"><font color='red'>org.ensembl.healthcheck.testcase.generic.CompareSchema</font></a></td><td><font color='green'>0</font></td><td><font color='red'>3</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.CoreForeignKeys"><font color='green'>org.ensembl.healthcheck.testcase.generic.CoreForeignKeys</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DNAEmpty"><font color='green'>org.ensembl.healthcheck.testcase.generic.DNAEmpty</font></a></td><td><font color='green'>3</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies"><font color='red'>org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DensityFeatures"><font color='red'>org.ensembl.healthcheck.testcase.generic.DensityFeatures</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayLabels"><font color='green'>org.ensembl.healthcheck.testcase.generic.DisplayLabels</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms"><font color='green'>org.ensembl.healthcheck.testcase.generic.DisplayMarkerSynonyms</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DisplayXref"><font color='green'>org.ensembl.healthcheck.testcase.generic.DisplayXref</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateAssembly"><font color='red'>org.ensembl.healthcheck.testcase.generic.DuplicateAssembly</font></a></td><td><font color='green'>1</font></td><td><font color='red'>3</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateExons"><font color='red'>org.ensembl.healthcheck.testcase.generic.DuplicateExons</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.DuplicateGenes"><font color='red'>org.ensembl.healthcheck.testcase.generic.DuplicateGenes</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ESTStableID"><font color='green'>org.ensembl.healthcheck.testcase.generic.ESTStableID</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ESTStableIDs"><font color='green'>org.ensembl.healthcheck.testcase.generic.ESTStableIDs</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.EmptyTables"><font color='red'>org.ensembl.healthcheck.testcase.generic.EmptyTables</font></a></td><td><font color='green'>2</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExonStrandOrder"><font color='green'>org.ensembl.healthcheck.testcase.generic.ExonStrandOrder</font></a></td><td><font color='green'>3</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd"><font color='green'>org.ensembl.healthcheck.testcase.generic.ExonTranscriptStartEnd</font></a></td><td><font color='green'>3</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName"><font color='green'>org.ensembl.healthcheck.testcase.generic.ExternalDBDisplayName</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.FeatureAnalysis"><font color='red'>org.ensembl.healthcheck.testcase.generic.FeatureAnalysis</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.FeatureCoords"><font color='red'>org.ensembl.healthcheck.testcase.generic.FeatureCoords</font></a></td><td><font color='green'>3</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GOXrefs"><font color='green'>org.ensembl.healthcheck.testcase.generic.GOXrefs</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GeneStatus"><font color='red'>org.ensembl.healthcheck.testcase.generic.GeneStatus</font></a></td><td><font color='green'>1</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.GeneType"><font color='red'>org.ensembl.healthcheck.testcase.generic.GeneType</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.HumanCCDS"><font color='green'>org.ensembl.healthcheck.testcase.generic.HumanCCDS</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.InterproDescriptions"><font color='green'>org.ensembl.healthcheck.testcase.generic.InterproDescriptions</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Karyotype"><font color='green'>org.ensembl.healthcheck.testcase.generic.Karyotype</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable"><font color='red'>org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable</font></a></td><td><font color='green'>2</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MappingSession"><font color='red'>org.ensembl.healthcheck.testcase.generic.MappingSession</font></a></td><td><font color='green'>0</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MarkerFeatures"><font color='green'>org.ensembl.healthcheck.testcase.generic.MarkerFeatures</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Meta"><font color='red'>org.ensembl.healthcheck.testcase.generic.Meta</font></a></td><td><font color='green'>2</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MetaCoord"><font color='green'>org.ensembl.healthcheck.testcase.generic.MetaCoord</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies"><font color='green'>org.ensembl.healthcheck.testcase.generic.MetaCrossSpecies</font></a></td><td><font color='green'>1</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.OligoMismatches"><font color='green'>org.ensembl.healthcheck.testcase.generic.OligoMismatches</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.OligoXrefs"><font color='red'>org.ensembl.healthcheck.testcase.generic.OligoXrefs</font></a></td><td><font color='green'>0</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.PredictedXrefs"><font color='green'>org.ensembl.healthcheck.testcase.generic.PredictedXrefs</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation"><font color='red'>org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinFeatures"><font color='green'>org.ensembl.healthcheck.testcase.generic.ProteinFeatures</font></a></td><td><font color='green'>3</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion"><font color='green'>org.ensembl.healthcheck.testcase.generic.ProteinIDXrefsNoVersion</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Pseudogene"><font color='green'>org.ensembl.healthcheck.testcase.generic.Pseudogene</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.RepeatConsensus"><font color='green'>org.ensembl.healthcheck.testcase.generic.RepeatConsensus</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.RepeatFeature"><font color='green'>org.ensembl.healthcheck.testcase.generic.RepeatFeature</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies"><font color='red'>org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies</font></a></td><td><font color='green'>0</font></td><td><font color='red'>2</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem"><font color='red'>org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem</font></a></td><td><font color='green'>0</font></td><td><font color='red'>4</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionName"><font color='green'>org.ensembl.healthcheck.testcase.generic.SeqRegionName</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel"><font color='red'>org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel</font></a></td><td><font color='green'>3</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.SingleDBCollations"><font color='green'>org.ensembl.healthcheck.testcase.generic.SingleDBCollations</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.StableID"><font color='red'>org.ensembl.healthcheck.testcase.generic.StableID</font></a></td><td><font color='green'>1</font></td><td><font color='red'>1</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.Strand"><font color='green'>org.ensembl.healthcheck.testcase.generic.Strand</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate"><font color='green'>org.ensembl.healthcheck.testcase.generic.TranscriptsTranslate</font></a></td><td><font color='green'>2</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEnd"><font color='green'>org.ensembl.healthcheck.testcase.generic.TranslationStartEnd</font></a></td><td><font color='green'>3</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon"><font color='green'>org.ensembl.healthcheck.testcase.generic.TranslationStartEndExon</font></a></td><td><font color='green'>3</font></td><td><font color='red'>0</font></td></tr>
<tr><td><a href="#org.ensembl.healthcheck.testcase.generic.XrefTypes"><font color='green'>org.ensembl.healthcheck.testcase.generic.XrefTypes</font></a></td><td><font color='green'>4</font></td><td><font color='red'>0</font></td></tr>
</table></p>
<hr>
<h2>Detailed reports by database</h2>
<h3 class='boxed'><a name="homo_sapiens_cdna_38_36">homo_sapiens_cdna_38_36</a></h3>
<p>
<a name="homo_sapiens_cdna_38_36:BigGeneExon"></a> <font color='grey'><strong>BigGeneExon: </strong>105 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
</p><p>
<a name="homo_sapiens_cdna_38_36:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>No genes have status KNOWN</font><br>
</p><p>
<a name="homo_sapiens_cdna_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>Co-ordinate systems contig:null and clone:null have 27093 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="homo_sapiens_cdna_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>24466 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="homo_sapiens_cdna_38_36:DuplicateAssembly"></a> <font color='red'><strong>DuplicateAssembly: </strong>At least 35 duplicate rows in assembly table</font><br>
</p>
<h3 class='boxed'><a name="homo_sapiens_core_38_36">homo_sapiens_core_38_36</a></h3>
<p>
<a name="homo_sapiens_core_38_36:BigGeneExon"></a> <font color='grey'><strong>BigGeneExon: </strong>41 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
<a name="homo_sapiens_core_38_36:BigGeneExon"></a> <font color='red'><strong>BigGeneExon: </strong>1 gene is longer than 3000000 bases</font><br>
<a name="homo_sapiens_core_38_36:BigGeneExon"></a> <font color='red'><strong>BigGeneExon: </strong>Gene ENSG00000173409 (ARV1) KNOWN has length 3346985 (ARV1 homolog [Source:RefSeq_peptide;Acc:NP_073623])</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Sum of values for snpDensity from density_feature (37691) doesn't agree with value from seq_region_attrib (36259) for Y</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 287038 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 287253 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 286263 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 286404 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 286458 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 286815 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 286853 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 287389 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 287434 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 287390 is duplicated.</font><br>
<a name="homo_sapiens_core_38_36:DuplicateExons"></a> <font color='red'><strong>DuplicateExons: </strong>Has at least 1007 duplicated exons.</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000195440 (miRNA ID 35649) is duplicated - see gene ENSG00000199086 (miRNA ID 35653)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000199113 (miRNA ID 35712) is duplicated - see gene ENSG00000190946 (miRNA ID 35703)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000191406 (miRNA ID 36060) is duplicated - see gene ENSG00000199164 (miRNA ID 36067)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000191449 (snoRNA ID 38635) is duplicated - see gene ENSG00000201836 (snoRNA ID 38667)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000191908 (snoRNA ID 38696) is duplicated - see gene ENSG00000201539 (snoRNA ID 38707)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000201018 (snRNA ID 41775) is duplicated - see gene ENSG00000190505 (snRNA ID 41772)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000193227 (miRNA ID 42213) is duplicated - see gene ENSG00000199189 (miRNA ID 42224)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000198978 (miRNA ID 42247) is duplicated - see gene ENSG00000192756 (miRNA ID 42233)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene ENSG00000189654 (miRNA ID 42267) is duplicated - see gene ENSG00000198970 (miRNA ID 42281)</font><br>
<a name="homo_sapiens_core_38_36:DuplicateGenes"></a> <font color='red'><strong>DuplicateGenes: </strong>Has 302 duplicated genes.</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>gene_attrib has zero rows</font><br>
<a name="homo_sapiens_core_38_36:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>regulatory_factor_coding has zero rows</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 19517 length 2092375</font><br>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 19766 length 2089393</font><br>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 22137 length 2298739</font><br>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 22141 length 2298739</font><br>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 22246 length 2054681</font><br>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 22285 length 2054111</font><br>
<a name="homo_sapiens_core_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 47836 length 3346927</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 32 is used in transcript as well as gene</font><br>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 53 is used in transcript as well as gene</font><br>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 54 is used in transcript as well as gene</font><br>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 5081 is used in transcript as well as gene</font><br>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 5038, logic name selenocysteine is not used in any feature table</font><br>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 5079, logic name miRNA is not used in any feature table</font><br>
<a name="homo_sapiens_core_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 5100, logic name cisred_search is not used in any feature table</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:FeatureCoords"></a> <font color='red'><strong>FeatureCoords: </strong>4 rows in density_feature have seq_region_end > length in seq_region_table</font><br>
<a name="homo_sapiens_core_38_36:FeatureCoords"></a> <font color='red'><strong>FeatureCoords: </strong>19 rows in regulatory_feature have seq_region_end > length in seq_region_table</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>ensembl_ncRNA (miRNA) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>miRNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>ensembl_ncRNA (misc_RNA) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>misc_RNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_rRNA is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_tRNA is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>Mt_tRNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>ensembl_havana_gene (protein_coding) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>havana (protein_coding) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>ensembl_ncRNA (rRNA) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>rRNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>scRNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>ensembl_ncRNA (snoRNA) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>snoRNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>ensembl_ncRNA (snRNA) is not a valid analysis logic name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>snRNA_pseudogene is not a valid gene type name.</font><br>
<a name="homo_sapiens_core_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>tRNA_pseudogene is not a valid gene type name.</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table gene has features with logic name havana which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table gene has features with logic name ensembl_havana_gene which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table protein_feature has features with logic name Smart which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table protein_feature has features with logic name PIRSF which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table protein_feature has features with logic name Tigrfam which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table oligo_feature has features with logic name AlignAffy which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table transcript has features with logic name havana which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Feature table transcript has features with logic name ensembl_havana_transcript which will not be drawn</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>protein_feature analysis with analysis_id = 12 has unknown gff_source = 'PROSITEPATTERN'</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>protein_feature analysis with analysis_id = 5091 has unknown gff_source = 'SMART'</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>protein_feature analysis with analysis_id = 5092 has unknown gff_source = 'PIRSF'</font><br>
<a name="homo_sapiens_core_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>protein_feature analysis with analysis_id = 5095 has unknown gff_source = 'TIGRFAM'</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:MappingSession"></a> <font color='red'><strong>MappingSession: </strong>Old/new names homo_sapiens_core_37_35j homo_sapiens_core_27_35a do not chain properly</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:Meta"></a> <font color='red'><strong>Meta: </strong>assembly.default from meta table is NCBI36 but lowest ranked coordinate system has version NCBI35</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:OligoXrefs"></a> <font color='red'><strong>OligoXrefs: </strong>Database has more than 50 seq_regions in 'chromosome' coordinate system (actually 140) - test skipped</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:ProteinFeatureTranslation"></a> <font color='red'><strong>ProteinFeatureTranslation: </strong>protein_feature table has 549 features that are longer than the translation</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>Co-ordinate systems chromosome:NCBI36 and chromosome:NCBI35 have 25 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="homo_sapiens_core_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>36 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 1 rows in pm2_misc_set but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 44 rows in pm2_misc_feature_misc_set but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 27011 rows in chr2contig but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 88955 rows in misc_feature_misc_set but homo_sapiens_core_38_36 only has 44</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 441974 rows in misc_attrib but homo_sapiens_core_38_36 only has 132</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 132 rows in pm2_misc_attrib but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 27173 rows in clone2contig but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 44 rows in pm2_misc_feature but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 6 rows in misc_set but homo_sapiens_core_38_36 only has 1</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 26986 rows in sc2contig but homo_sapiens_core_38_36 has none</font><br>
<a name="homo_sapiens_core_38_36:ComparePreviousVersionTableRows"></a> <font color='red'><strong>ComparePreviousVersionTableRows: </strong>homo_sapiens_core_38_36 has 88955 rows in misc_feature but homo_sapiens_core_38_36 only has 44</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:AssemblyTablesAcrossSpecies"></a> <font color='red'><strong>AssemblyTablesAcrossSpecies: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 92972, 83479</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:AttribTypeAcrossSpecies"></a> <font color='red'><strong>AttribTypeAcrossSpecies: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 70, 58</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table assembly_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_core_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table coord_system_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_core_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table meta_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_core_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table oligo_feature_backup exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_core_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table seq_region_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_core_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table name discrepancy detected, skipping rest of checks</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:DNATablesAcrossSpecies"></a> <font color='red'><strong>DNATablesAcrossSpecies: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 27093, 0</font><br>
</p><p>
<a name="homo_sapiens_core_38_36:SeqRegionAttribAcrossSpecies"></a> <font color='red'><strong>SeqRegionAttribAcrossSpecies: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 816, 237</font><br>
</p>
<h3 class='boxed'><a name="homo_sapiens_otherfeatures_38_36">homo_sapiens_otherfeatures_38_36</a></h3>
<p>
<a name="homo_sapiens_otherfeatures_38_36:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>No genes have status KNOWN</font><br>
<a name="homo_sapiens_otherfeatures_38_36:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>28639 genes have null status</font><br>
<a name="homo_sapiens_otherfeatures_38_36:GeneStatus"></a> <font color='red'><strong>GeneStatus: </strong>58916 transcripts have null status</font><br>
</p><p>
<a name="homo_sapiens_otherfeatures_38_36:GeneType"></a> <font color='red'><strong>GeneType: </strong>genomewise (protein_coding) is not a valid analysis logic name.</font><br>
</p><p>
<a name="homo_sapiens_otherfeatures_38_36:LogicNamesDisplayable"></a> <font color='red'><strong>LogicNamesDisplayable: </strong>Analysis with logic name 'sheep_bac_ends' has a blank db field - features of this type will have no label in ProtView</font><br>
</p><p>
<a name="homo_sapiens_otherfeatures_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>Co-ordinate systems contig:null and clone:null have 27093 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="homo_sapiens_otherfeatures_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>24466 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="homo_sapiens_otherfeatures_38_36:DuplicateAssembly"></a> <font color='red'><strong>DuplicateAssembly: </strong>At least 35 duplicate rows in assembly table</font><br>
</p>
<h3 class='boxed'><a name="homo_sapiens_vega_38_36">homo_sapiens_vega_38_36</a></h3>
<p>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 1 (ID 226034) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 10 (ID 226043) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 11 (ID 226044) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 12 (ID 226056) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 13 (ID 226050) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 14 (ID 226039) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 15 (ID 226057) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 16 (ID 226036) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 17 (ID 226033) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 18 (ID 226035) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 19 (ID 226061) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 2 (ID 226032) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 20 (ID 226040) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 21 (ID 226028) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 22 (ID 226046) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 3 (ID 226053) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 4 (ID 226060) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 5 (ID 226062) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 6 (ID 226052) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 7 (ID 226030) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 8 (ID 226058) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region 9 (ID 226055) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region MT (ID 226038) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region X (ID 226054) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>Top-level seq region Y (ID 226031) has no density features</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>No entry in density_type for analysis PercentGC</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>No entry in density_type for analysis PercentageRepeat</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>No entry in density_type for analysis knownGeneDensity</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>No entry in density_type for analysis geneDensity</font><br>
<a name="homo_sapiens_vega_38_36:DensityFeatures"></a> <font color='red'><strong>DensityFeatures: </strong>No entry in density_type for analysis snpDensity</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 197264 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 197444 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77796 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77790 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77795 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77789 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77794 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77788 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77787 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='black'><strong>DuplicateExons: </strong>Exon 77816 is duplicated.</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateExons"></a> <font color='red'><strong>DuplicateExons: </strong>Has at least 962 duplicated exons.</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000023670 (protein_coding ID 7182) is duplicated - see gene OTTHUMG00000119091 (protein_coding ID 7183)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000073451 (protein_coding ID 15748) is duplicated - see gene OTTHUMG00000128707 (protein_coding ID 15749)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000090677 (protein_coding ID 15956) is duplicated - see gene OTTHUMG00000072969 (protein_coding ID 15955)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000048082 (protein_coding ID 16028) is duplicated - see gene OTTHUMG00000073578 (protein_coding ID 16027)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000090497 (protein_coding ID 15812) is duplicated - see gene OTTHUMG00000073463 (protein_coding ID 15811)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000073467 (protein_coding ID 15966) is duplicated - see gene OTTHUMG00000090721 (protein_coding ID 15967)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='black'><strong>DuplicateGenes: </strong>Gene OTTHUMG00000014526 (processed_pseudogene ID 5292) is duplicated - see gene OTTHUMG00000078409 (processed_pseudogene ID 5294)</font><br>
<a name="homo_sapiens_vega_38_36:DuplicateGenes"></a> <font color='red'><strong>DuplicateGenes: </strong>Has 7 duplicated genes.</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>oligo_array has zero rows</font><br>
<a name="homo_sapiens_vega_38_36:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>oligo_feature has zero rows</font><br>
<a name="homo_sapiens_vega_38_36:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>oligo_probe has zero rows</font><br>
<a name="homo_sapiens_vega_38_36:EmptyTables"></a> <font color='red'><strong>EmptyTables: </strong>regulatory_search_region has zero rows</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 20375 length 2299488</font><br>
<a name="homo_sapiens_vega_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 20933 length 2084571</font><br>
<a name="homo_sapiens_vega_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 41919 length 2054681</font><br>
<a name="homo_sapiens_vega_38_36:ExonStrandOrder"></a> <font color='black'><strong>ExonStrandOrder: </strong>Long transcript 46162 length 2092375</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 1 is used in transcript as well as gene</font><br>
<a name="homo_sapiens_vega_38_36:FeatureAnalysis"></a> <font color='red'><strong>FeatureAnalysis: </strong>Analysis with ID 15, logic name otter_corf is not used in any feature table</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:Meta"></a> <font color='red'><strong>Meta: </strong>assembly.default from meta table is NCBI36 but lowest ranked coordinate system has version VEGA</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:ProteinFeatureTranslation"></a> <font color='red'><strong>ProteinFeatureTranslation: </strong>protein_feature table has 11 features that are longer than the translation</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>Co-ordinate systems chromosome:NCBI36 and chromosome:VEGA have 24 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>Co-ordinate systems contig:null and clone:null have 27093 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionCoordSystem"></a> <font color='red'><strong>SeqRegionCoordSystem: </strong>24490 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226028</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226030</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226031</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226032</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226034</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226035</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226036</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226039</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226040</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226043</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226044</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226046</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226050</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226052</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226054</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226055</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226056</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226057</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226058</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226060</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226061</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No top_level attribute set for seq_region_id 226062</font><br>
<a name="homo_sapiens_vega_38_36:SeqRegionsTopLevel"></a> <font color='red'><strong>SeqRegionsTopLevel: </strong>No seq_regions are marked as toplevel. This may cause problems for Compara</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:StableID"></a> <font color='red'><strong>StableID: </strong>exon_stable_id has 736 duplicate stable IDs (versions not checked)</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:DuplicateAssembly"></a> <font color='red'><strong>DuplicateAssembly: </strong>At least 35 duplicate rows in assembly table</font><br>
</p><p>
<a name="homo_sapiens_vega_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table tmp_assembly exists in homo_sapiens_vega_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_vega_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table tmp_seq_region exists in homo_sapiens_vega_38_36 but not in master_schema_38</font><br>
<a name="homo_sapiens_vega_38_36:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>Table name discrepancy detected, skipping rest of checks</font><br>
</p>
<h3 class='boxed'><a name="master_schema_38">master_schema_38</a></h3>
<p>
<a name="master_schema_38:CompareSchema"></a> <font color='red'><strong>CompareSchema: </strong>master_schema_38 and homo_sapiens_otherfeatures_38_36 oligo_feature differ at row 8 column 2 (Type) Values: int(10) unsigned, int(11)</font><br>
</p>
<hr>
<h2>Detailed reports by test case</h2>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies">org.ensembl.healthcheck.testcase.generic.AssemblyTablesAcrossSpecies</a></h3>
<p>
<a name="AssemblyTablesAcrossSpecies:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 92972, 83479</font><br>
</p><p>
<a name="AssemblyTablesAcrossSpecies:"></a> <font color='red'><strong>: </strong>Differences in assembly table across species</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies">org.ensembl.healthcheck.testcase.generic.AttribTypeAcrossSpecies</a></h3>
<p>
<a name="AttribTypeAcrossSpecies:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 70, 58</font><br>
</p><p>
<a name="AttribTypeAcrossSpecies:"></a> <font color='red'><strong>: </strong>Differences in attrib_type table across species</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.BigGeneExon">org.ensembl.healthcheck.testcase.generic.BigGeneExon</a></h3>
<p>
<a name="BigGeneExon:homo_sapiens_cdna_38_36"></a> <font color='grey'><strong>homo_sapiens_cdna_38_36: </strong>105 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
</p><p>
<a name="BigGeneExon:homo_sapiens_core_38_36"></a> <font color='grey'><strong>homo_sapiens_core_38_36: </strong>41 genes are longer than 1000000 bases but less than 3000000 bases</font><br>
<a name="BigGeneExon:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>1 gene is longer than 3000000 bases</font><br>
<a name="BigGeneExon:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000173409 (ARV1) KNOWN has length 3346985 (ARV1 homolog [Source:RefSeq_peptide;Acc:NP_073623])</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows">org.ensembl.healthcheck.testcase.generic.ComparePreviousVersionTableRows</a></h3>
<p>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 1 rows in pm2_misc_set but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 44 rows in pm2_misc_feature_misc_set but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 27011 rows in chr2contig but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 88955 rows in misc_feature_misc_set but homo_sapiens_core_38_36 only has 44</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 441974 rows in misc_attrib but homo_sapiens_core_38_36 only has 132</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 132 rows in pm2_misc_attrib but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 27173 rows in clone2contig but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 44 rows in pm2_misc_feature but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 6 rows in misc_set but homo_sapiens_core_38_36 only has 1</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 26986 rows in sc2contig but homo_sapiens_core_38_36 has none</font><br>
<a name="ComparePreviousVersionTableRows:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 has 88955 rows in misc_feature but homo_sapiens_core_38_36 only has 44</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.CompareSchema">org.ensembl.healthcheck.testcase.generic.CompareSchema</a></h3>
<p>
<a name="CompareSchema:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Table assembly_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Table coord_system_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Table meta_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Table oligo_feature_backup exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Table seq_region_bak exists in homo_sapiens_core_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Table name discrepancy detected, skipping rest of checks</font><br>
</p><p>
<a name="CompareSchema:master_schema_38"></a> <font color='red'><strong>master_schema_38: </strong>master_schema_38 and homo_sapiens_otherfeatures_38_36 oligo_feature differ at row 8 column 2 (Type) Values: int(10) unsigned, int(11)</font><br>
</p><p>
<a name="CompareSchema:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Table tmp_assembly exists in homo_sapiens_vega_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Table tmp_seq_region exists in homo_sapiens_vega_38_36 but not in master_schema_38</font><br>
<a name="CompareSchema:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Table name discrepancy detected, skipping rest of checks</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies">org.ensembl.healthcheck.testcase.generic.DNATablesAcrossSpecies</a></h3>
<p>
<a name="DNATablesAcrossSpecies:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 27093, 0</font><br>
</p><p>
<a name="DNATablesAcrossSpecies:"></a> <font color='red'><strong>: </strong>Differences in dna table across species</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DensityFeatures">org.ensembl.healthcheck.testcase.generic.DensityFeatures</a></h3>
<p>
<a name="DensityFeatures:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Sum of values for snpDensity from density_feature (37691) doesn't agree with value from seq_region_attrib (36259) for Y</font><br>
</p><p>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 1 (ID 226034) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 10 (ID 226043) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 11 (ID 226044) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 12 (ID 226056) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 13 (ID 226050) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 14 (ID 226039) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 15 (ID 226057) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 16 (ID 226036) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 17 (ID 226033) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 18 (ID 226035) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 19 (ID 226061) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 2 (ID 226032) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 20 (ID 226040) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 21 (ID 226028) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 22 (ID 226046) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 3 (ID 226053) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 4 (ID 226060) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 5 (ID 226062) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 6 (ID 226052) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 7 (ID 226030) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 8 (ID 226058) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region 9 (ID 226055) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region MT (ID 226038) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region X (ID 226054) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Top-level seq region Y (ID 226031) has no density features</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No entry in density_type for analysis PercentGC</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No entry in density_type for analysis PercentageRepeat</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No entry in density_type for analysis knownGeneDensity</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No entry in density_type for analysis geneDensity</font><br>
<a name="DensityFeatures:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No entry in density_type for analysis snpDensity</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DuplicateAssembly">org.ensembl.healthcheck.testcase.generic.DuplicateAssembly</a></h3>
<p>
<a name="DuplicateAssembly:homo_sapiens_cdna_38_36"></a> <font color='red'><strong>homo_sapiens_cdna_38_36: </strong>At least 35 duplicate rows in assembly table</font><br>
</p><p>
<a name="DuplicateAssembly:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>At least 35 duplicate rows in assembly table</font><br>
</p><p>
<a name="DuplicateAssembly:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>At least 35 duplicate rows in assembly table</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DuplicateExons">org.ensembl.healthcheck.testcase.generic.DuplicateExons</a></h3>
<p>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 287038 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 287253 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 286263 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 286404 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 286458 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 286815 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 286853 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 287389 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 287434 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Exon 287390 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Has at least 1007 duplicated exons.</font><br>
</p><p>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 197264 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 197444 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77796 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77790 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77795 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77789 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77794 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77788 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77787 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Exon 77816 is duplicated.</font><br>
<a name="DuplicateExons:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Has at least 962 duplicated exons.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.DuplicateGenes">org.ensembl.healthcheck.testcase.generic.DuplicateGenes</a></h3>
<p>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000195440 (miRNA ID 35649) is duplicated - see gene ENSG00000199086 (miRNA ID 35653)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000199113 (miRNA ID 35712) is duplicated - see gene ENSG00000190946 (miRNA ID 35703)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000191406 (miRNA ID 36060) is duplicated - see gene ENSG00000199164 (miRNA ID 36067)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000191449 (snoRNA ID 38635) is duplicated - see gene ENSG00000201836 (snoRNA ID 38667)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000191908 (snoRNA ID 38696) is duplicated - see gene ENSG00000201539 (snoRNA ID 38707)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000201018 (snRNA ID 41775) is duplicated - see gene ENSG00000190505 (snRNA ID 41772)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000193227 (miRNA ID 42213) is duplicated - see gene ENSG00000199189 (miRNA ID 42224)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000198978 (miRNA ID 42247) is duplicated - see gene ENSG00000192756 (miRNA ID 42233)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Gene ENSG00000189654 (miRNA ID 42267) is duplicated - see gene ENSG00000198970 (miRNA ID 42281)</font><br>
<a name="DuplicateGenes:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Has 302 duplicated genes.</font><br>
</p><p>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000023670 (protein_coding ID 7182) is duplicated - see gene OTTHUMG00000119091 (protein_coding ID 7183)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000073451 (protein_coding ID 15748) is duplicated - see gene OTTHUMG00000128707 (protein_coding ID 15749)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000090677 (protein_coding ID 15956) is duplicated - see gene OTTHUMG00000072969 (protein_coding ID 15955)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000048082 (protein_coding ID 16028) is duplicated - see gene OTTHUMG00000073578 (protein_coding ID 16027)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000090497 (protein_coding ID 15812) is duplicated - see gene OTTHUMG00000073463 (protein_coding ID 15811)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000073467 (protein_coding ID 15966) is duplicated - see gene OTTHUMG00000090721 (protein_coding ID 15967)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Gene OTTHUMG00000014526 (processed_pseudogene ID 5292) is duplicated - see gene OTTHUMG00000078409 (processed_pseudogene ID 5294)</font><br>
<a name="DuplicateGenes:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Has 7 duplicated genes.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.EmptyTables">org.ensembl.healthcheck.testcase.generic.EmptyTables</a></h3>
<p>
<a name="EmptyTables:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>gene_attrib has zero rows</font><br>
<a name="EmptyTables:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>regulatory_factor_coding has zero rows</font><br>
</p><p>
<a name="EmptyTables:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>oligo_array has zero rows</font><br>
<a name="EmptyTables:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>oligo_feature has zero rows</font><br>
<a name="EmptyTables:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>oligo_probe has zero rows</font><br>
<a name="EmptyTables:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>regulatory_search_region has zero rows</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ExonStrandOrder">org.ensembl.healthcheck.testcase.generic.ExonStrandOrder</a></h3>
<p>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 19517 length 2092375</font><br>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 19766 length 2089393</font><br>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 22137 length 2298739</font><br>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 22141 length 2298739</font><br>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 22246 length 2054681</font><br>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 22285 length 2054111</font><br>
<a name="ExonStrandOrder:homo_sapiens_core_38_36"></a> <font color='black'><strong>homo_sapiens_core_38_36: </strong>Long transcript 47836 length 3346927</font><br>
</p><p>
<a name="ExonStrandOrder:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Long transcript 20375 length 2299488</font><br>
<a name="ExonStrandOrder:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Long transcript 20933 length 2084571</font><br>
<a name="ExonStrandOrder:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Long transcript 41919 length 2054681</font><br>
<a name="ExonStrandOrder:homo_sapiens_vega_38_36"></a> <font color='black'><strong>homo_sapiens_vega_38_36: </strong>Long transcript 46162 length 2092375</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.FeatureAnalysis">org.ensembl.healthcheck.testcase.generic.FeatureAnalysis</a></h3>
<p>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 32 is used in transcript as well as gene</font><br>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 53 is used in transcript as well as gene</font><br>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 54 is used in transcript as well as gene</font><br>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 5081 is used in transcript as well as gene</font><br>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 5038, logic name selenocysteine is not used in any feature table</font><br>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 5079, logic name miRNA is not used in any feature table</font><br>
<a name="FeatureAnalysis:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Analysis with ID 5100, logic name cisred_search is not used in any feature table</font><br>
</p><p>
<a name="FeatureAnalysis:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Analysis with ID 1 is used in transcript as well as gene</font><br>
<a name="FeatureAnalysis:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Analysis with ID 15, logic name otter_corf is not used in any feature table</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.FeatureCoords">org.ensembl.healthcheck.testcase.generic.FeatureCoords</a></h3>
<p>
<a name="FeatureCoords:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>4 rows in density_feature have seq_region_end > length in seq_region_table</font><br>
<a name="FeatureCoords:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>19 rows in regulatory_feature have seq_region_end > length in seq_region_table</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.GeneStatus">org.ensembl.healthcheck.testcase.generic.GeneStatus</a></h3>
<p>
<a name="GeneStatus:homo_sapiens_cdna_38_36"></a> <font color='red'><strong>homo_sapiens_cdna_38_36: </strong>No genes have status KNOWN</font><br>
</p><p>
<a name="GeneStatus:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>No genes have status KNOWN</font><br>
<a name="GeneStatus:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>28639 genes have null status</font><br>
<a name="GeneStatus:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>58916 transcripts have null status</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.GeneType">org.ensembl.healthcheck.testcase.generic.GeneType</a></h3>
<p>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>ensembl_ncRNA (miRNA) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>miRNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>ensembl_ncRNA (misc_RNA) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>misc_RNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Mt_rRNA is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Mt_tRNA is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Mt_tRNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>ensembl_havana_gene (protein_coding) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>havana (protein_coding) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>ensembl_ncRNA (rRNA) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>rRNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>scRNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>ensembl_ncRNA (snoRNA) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>snoRNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>ensembl_ncRNA (snRNA) is not a valid analysis logic name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>snRNA_pseudogene is not a valid gene type name.</font><br>
<a name="GeneType:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>tRNA_pseudogene is not a valid gene type name.</font><br>
</p><p>
<a name="GeneType:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>genomewise (protein_coding) is not a valid analysis logic name.</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable">org.ensembl.healthcheck.testcase.generic.LogicNamesDisplayable</a></h3>
<p>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table gene has features with logic name havana which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table gene has features with logic name ensembl_havana_gene which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table protein_feature has features with logic name Smart which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table protein_feature has features with logic name PIRSF which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table protein_feature has features with logic name Tigrfam which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table oligo_feature has features with logic name AlignAffy which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table transcript has features with logic name havana which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Feature table transcript has features with logic name ensembl_havana_transcript which will not be drawn</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>protein_feature analysis with analysis_id = 12 has unknown gff_source = 'PROSITEPATTERN'</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>protein_feature analysis with analysis_id = 5091 has unknown gff_source = 'SMART'</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>protein_feature analysis with analysis_id = 5092 has unknown gff_source = 'PIRSF'</font><br>
<a name="LogicNamesDisplayable:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>protein_feature analysis with analysis_id = 5095 has unknown gff_source = 'TIGRFAM'</font><br>
</p><p>
<a name="LogicNamesDisplayable:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>Analysis with logic name 'sheep_bac_ends' has a blank db field - features of this type will have no label in ProtView</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.MappingSession">org.ensembl.healthcheck.testcase.generic.MappingSession</a></h3>
<p>
<a name="MappingSession:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Old/new names homo_sapiens_core_37_35j homo_sapiens_core_27_35a do not chain properly</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.Meta">org.ensembl.healthcheck.testcase.generic.Meta</a></h3>
<p>
<a name="Meta:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>assembly.default from meta table is NCBI36 but lowest ranked coordinate system has version NCBI35</font><br>
</p><p>
<a name="Meta:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>assembly.default from meta table is NCBI36 but lowest ranked coordinate system has version VEGA</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.OligoXrefs">org.ensembl.healthcheck.testcase.generic.OligoXrefs</a></h3>
<p>
<a name="OligoXrefs:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Database has more than 50 seq_regions in 'chromosome' coordinate system (actually 140) - test skipped</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation">org.ensembl.healthcheck.testcase.generic.ProteinFeatureTranslation</a></h3>
<p>
<a name="ProteinFeatureTranslation:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>protein_feature table has 549 features that are longer than the translation</font><br>
</p><p>
<a name="ProteinFeatureTranslation:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>protein_feature table has 11 features that are longer than the translation</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies">org.ensembl.healthcheck.testcase.generic.SeqRegionAttribAcrossSpecies</a></h3>
<p>
<a name="SeqRegionAttribAcrossSpecies:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>homo_sapiens_core_38_36 and homo_sapiens_vega_38_36  differ at row 1 column 1 (COUNT(*)) Values: 816, 237</font><br>
</p><p>
<a name="SeqRegionAttribAcrossSpecies:"></a> <font color='red'><strong>: </strong>Differences in seq_region_attrib table across species</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem">org.ensembl.healthcheck.testcase.generic.SeqRegionCoordSystem</a></h3>
<p>
<a name="SeqRegionCoordSystem:homo_sapiens_cdna_38_36"></a> <font color='red'><strong>homo_sapiens_cdna_38_36: </strong>Co-ordinate systems contig:null and clone:null have 27093 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="SeqRegionCoordSystem:homo_sapiens_cdna_38_36"></a> <font color='red'><strong>homo_sapiens_cdna_38_36: </strong>24466 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="SeqRegionCoordSystem:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>Co-ordinate systems chromosome:NCBI36 and chromosome:NCBI35 have 25 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="SeqRegionCoordSystem:homo_sapiens_core_38_36"></a> <font color='red'><strong>homo_sapiens_core_38_36: </strong>36 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="SeqRegionCoordSystem:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>Co-ordinate systems contig:null and clone:null have 27093 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="SeqRegionCoordSystem:homo_sapiens_otherfeatures_38_36"></a> <font color='red'><strong>homo_sapiens_otherfeatures_38_36: </strong>24466 seq_regions have the same name but different lengths</font><br>
</p><p>
<a name="SeqRegionCoordSystem:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Co-ordinate systems chromosome:NCBI36 and chromosome:VEGA have 24 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="SeqRegionCoordSystem:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>Co-ordinate systems contig:null and clone:null have 27093 identically-named seq_regions - this may cause problems for ID mapping</font><br>
<a name="SeqRegionCoordSystem:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>24490 seq_regions have the same name but different lengths</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel">org.ensembl.healthcheck.testcase.generic.SeqRegionsTopLevel</a></h3>
<p>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226028</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226030</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226031</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226032</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226034</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226035</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226036</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226039</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226040</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226043</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226044</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226046</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226050</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226052</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226054</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226055</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226056</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226057</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226058</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226060</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226061</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No top_level attribute set for seq_region_id 226062</font><br>
<a name="SeqRegionsTopLevel:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>No seq_regions are marked as toplevel. This may cause problems for Compara</font><br>
</p>
<h3 class='boxed'><a name="org.ensembl.healthcheck.testcase.generic.StableID">org.ensembl.healthcheck.testcase.generic.StableID</a></h3>
<p>
<a name="StableID:homo_sapiens_vega_38_36"></a> <font color='red'><strong>homo_sapiens_vega_38_36: </strong>exon_stable_id has 736 duplicate stable IDs (versions not checked)</font><br>
</p>
<hr>
<p>Test run was started at Wed May 03 02:18:05 BST 2006 and finished at Wed May 03 03:00:51 BST 2006<br>
 Run time 42 min 45s </p>
<h4>Configuration used:</h4>
<pre>
Tests/groups run:   release<br>
Database host:      ecs2:3364<br>
Database names:     ^homo_sapiens_(core|otherfeatures|cdna|vega)_[0-9]+_.*<br>
Output file:        web_homo_sapiens.html<br>
Output level:       info<br>
</pre>
</div>
</body>
</html>
<hr>
